Dockerfile.rocm

```diff
@@ -1,17 +1,18 @@
 # default base image
-ARG BASE_IMAGE="rocm/pytorch:rocm6.1.2_ubuntu20.04_py3.9_pytorch_staging"
+ARG BASE_IMAGE="rocm/pytorch:rocm6.2_ubuntu20.04_py3.9_pytorch_staging"
```

Comment:
Any reason why we should use 22.04? Besides it being newer which does not bring any performance improvements.

Reply:
Because we can...?
I am doing all my testing on 22 lately and don't see any issues. But mainly yeah, more newer more better, 20 will be eventually deprecated as was 18 (next April to be precise)

Reply:
"Because we can" is not an affirmative reason. "More newer more better" is dubious at best.

Deprecation is something we can consider but not for ROCm 6.2 unless in the unlikely event there's a patch for it released after next April which we use.

I'll draw your attention to PRs which may have missed your attention like https://github.com/vllm-project/vllm/pull/6517 for one reason we should consider upgrades carefully. We have seen other limitations in Ubuntu 22.04 versions in the past, such as lack of prebuilt ASan libraries obtainable via `apt` on ROCm 6.1. Absent a affirmative reason, disturbing the status quo is fixing what isn't broke.